yellow-paper/docs/rollup-circuits/index.md

```diff
@@ -7,18 +7,25 @@ title: Rollup Circuits
 Together with the [validating light node](../l1-smart-contracts/index.md) the rollup circuits must ensure that incoming blocks are valid, that state is progressed correctly and that anyone can rebuild the state.
 
 To support this, we construct a single proof for the entire block, which is then verified by the validating light node. 
-This single proof is constructed by recursively merging proofs together in a binary tree structure. 
+This single proof is constructed by recursively merging proofs together in two tree subtrees, and then combine these use those trees at the root . 
 This structure allows us to keep the workload of each individual proof small, while making it very parallelizable. 
 This works very well for the case where we want many actors to be able to participate in the proof generation.
 
-The tree structure is outlined below, but the general idea is that we have a tree where all the leaves are transactions (kernel proofs) and through $\log(n)$ steps we can then "compress" them down to just a single root proof. 
-Note that we have two different types of "merger" circuit, namely:
+The tree structure is outlined below, but the general idea is that we have one subtree tree where all the leaves are transactions (kernel proofs) and through $\log(n)$ steps we can then "compress" them down to just a single root proof.
+And another tree where all the leaves are messages (cross-chain messages) that is also "compressed"  down to just a single proof going into the final root.
 
+Note that we have two different types of "merger" circuit, depending on what they are compressing.
+
+For transaction compression we have:
 - The merge rollup
   - Merges two base rollup proofs OR two merge rollup proofs
 - The root rollup
   - Merges two merge rollup proofs
 
+And for the message compression we have:
+- The compress circuit
```

Comment:
I feel like having merge circuit and compress circuit is not a great naming. Maybe something like msg_compress circuit and tx_compress circuits?

Reply:
Ye will look at finding some better names for it generally, also want to get rid of the "rollup" part as mentioned above.